src/source/Signaling/LwsApiCalls.c

```diff
@@ -1627,45 +1693,85 @@ STATUS receiveLwsMessage(PSignalingClient pSignalingClient, PCHAR pMessage, UINT
             pSignalingMessageWrapper->receivedSignalingMessage.signalingMessage.payload[MAX_SIGNALING_MESSAGE_LEN] = '\0';
             pSignalingMessageWrapper->receivedSignalingMessage.signalingMessage.payloadLen = outLen;
             i++;
-        } else {
-            if (!parsedStatusResponse) {
-                if (compareJsonString(pMessage, &tokens[i], JSMN_STRING, (PCHAR) "statusResponse")) {
-                    parsedStatusResponse = TRUE;
-                    i++;
-                }
-            } else {
-                if (compareJsonString(pMessage, &tokens[i], JSMN_STRING, (PCHAR) "correlationId")) {
-                    strLen = (UINT32)(tokens[i + 1].end - tokens[i + 1].start);
-                    CHK(strLen <= MAX_CORRELATION_ID_LEN, STATUS_INVALID_API_CALL_RETURN_JSON);
-                    STRNCPY(pSignalingMessageWrapper->receivedSignalingMessage.signalingMessage.correlationId, pMessage + tokens[i + 1].start,
-                            strLen);
-                    pSignalingMessageWrapper->receivedSignalingMessage.signalingMessage.correlationId[MAX_CORRELATION_ID_LEN] = '\0';
+        } else if (!parsedStatusResponse && compareJsonString(pMessage, &tokens[i], JSMN_STRING, (PCHAR) "statusResponse")) {
+            parsedStatusResponse = TRUE;
+            i++;
+        } else if (parsedStatusResponse && compareJsonString(pMessage, &tokens[i], JSMN_STRING, (PCHAR) "correlationId")) {
```

Comment:
Can we add the subsequent blocks this way? 

```
if (parsedStatusResponse) {
   if(compareJsonString....) {
   }
   else if (compareJsonString...) {
   }
   else if (compareJsonString...) {
   }
}
```

Reply:
won't work really as the JSON could contain response first and we will enter this clause for all of the consequent parsing for ICE which is wrong. I really need to think on how to simplify these JSON parsing routines further but not for this commit though